src/inventory/inventoryRepository.js

```diff
@@ -64,4 +65,69 @@ export class InventoryRepository {
       throw error;
     }
   }
+
+  async removeItemToUser (userName, item, quantity) {
+    try {
+      const rows = await this.mySQLConnection.executeQuery(
+        `
+      SELECT Quantity
+      FROM user_items
+      WHERE user_userName = ?
+      AND item_Name = ?;
+      `, [userName, item]
+      );
+      if (rows.length === 0) {
+        return false;
+      }
+      const actualQuantity = rows[0].Quantity;
+      if (actualQuantity < quantity) {
+        return false;
+      }
+      if (actualQuantity === quantity) {
+        const res = await this.mySQLConnection.executeQuery(
+          `
+        DELETE FROM user_items
+        WHERE user_userName = ?
+        AND item_Name = ?;
+        `, [userName, item]
+        );
+        return res.affectedRows > 0;
+      } else {
+        const updatedQuantity = actualQuantity - quantity;
+        const res = await this.mySQLConnection.executeQuery(
+          `
+        UPDATE user_items
+        SET Quantity = ?
+        WHERE user_userName = ?
+        AND item_Name = ?;
+        `, [updatedQuantity, userName, item]
+        );
+        return res.affectedRows > 0;
+      }
+    } catch (error) {
+      if (error.name === SQLERROR) {
+        throw new FailedToDeleteError(FAILED_DELETING, INVENTORY, error);
+      }
+      throw error;
+    }
+  }
+
+  async getQuantity (userName, name) {
+    try {
+      const rows = await this.mySQLConnection.executeQuery(
+        `
+        SELECT Quantity FROM user_items
+        WHERE user_userName = ? AND item_Name = ?
+        `, [userName, name]
+      );
+      if (!rows.length === 0) {
+        return false;
```

Comment:
Here you are returning false when the function is expected to return a number, if that item is not found the quantity you have is 0, you should return 0